tests/upgrade/test_crossgrade.sh

```diff
@@ -365,19 +370,19 @@ resetCluster() {
     echo_and_run_or_die kubectl label namespace "${TEST_NAMESPACE}" istio-injection=enabled
 }
 
-# Returns 0 if the passed string has form "Code 200 : 6601 (94.6 %)" and the percentage is smaller than ${MIN_200_PCT_FOR_PASS}
-percent200sAbove() {
-    local s=$1
-    local regex="Code 200 : [0-9]+ \\(([0-9]+)\\.[0-9]+ %\\)"
-    if [[ $s =~ $regex ]]; then
-        local pct200s="${BASH_REMATCH[1]}"
-        if (( pct200s > MIN_200_PCT_FOR_PASS )); then
-            return 0
-        fi
-        return 1
-    fi
-    echo "No Code 200 percentage found in log."
-    return 1
+# Return 1 if the specific error code percentage exceed corresponding threshold
+errorPercentBelow() {
+     local s
+     s=$(grep "${2}" "${1}")
+     local regex="${2} : [0-9]+ \\(([0-9]+)\\.[0-9]+ %\\)"
+     if [[ ${s} =~ ${regex} ]]; then
+          local pctErr="${BASH_REMATCH[1]}"
+          if (( pctErr > ${3} )); then
+             return 1
+          fi
+             echo "Errors percentage is within threshold"
+     fi
+     return 0
```

Comment:
If the the regex does not match, the function used to return 1 (error), but it is now returning 0 (success). Is this intended? Maybe error is a better default?

Reply:
Previously the function is to check whether 200 code percentage is larger than 85% so it returns 1 if there is match found. But here it is checking whether error codes are below threshold instead

Reply:
Makes sense.